src/game/Spell.cpp

```diff
@@ -6244,6 +6265,12 @@ bool Spell::CheckTarget(Unit* target, SpellEffectIndex eff)
     // A player can cast spells on his pet (or other controlled unit) though in any state
     if (target != m_caster && target->GetCharmerOrOwnerGuid() != m_caster->GetObjectGuid())
     {
+        switch (m_spellInfo->Id) // spells that target minipets, which are inherently non attackable
+        {
+            case 44877:
```

Comment:
spell have in field Targets
    TARGET_MINION                      = 32,
so no need horrible spell id hasks
Expect proper fill targtes list in Spell::SetTargetMap base at this value instead

        case TARGET_MINION:
            if (m_spellInfo->Effect[effIndex] != SPELL_EFFECT_DUEL)
                targetUnitMap.push_back(m_caster);
            break;
but ofc need check that current working spells with same field value not affected in wrong way